src/twitter/Timeline.js

```diff
@@ -296,35 +347,49 @@ class Timeline {
       this._newtags.delete("#"+feed.hashtag);
     }
 
-    this.twitter.storage.get_since(feed.hashtag).then((since) => {
-      log.silly("Polling %s, since value: %s", feed.hashtag, since);
-      if (since) {
-        req.since_id = since;
-      }
-      return this.twitter.client_factory.get_client();
-    }).then((client)=>{
-      return client.getAsync('search/tweets', req);
-    }).then((results) => {
-      if (results.statuses.length === 0) {
-        return;
+    const since = yield this.twitter.storage.get_since(feed.hashtag);
+    if (since) {
+      req.since_id = since;
+    }
+    let results;
+    try {
+      results = yield client.getAsync('search/tweets', req);
+    }
+    catch (error) {
+      log.error("Timeline", "_process_hashtags: GET /search/tweets returned: %s", error.code);
+      return;
+    }
+    if (results.statuses.length === 0) {
+      return;
+    }
+    else if(results.statuses.length === HASHTAG_TWEET_FETCH_COUNT) {
+      log.info("Timeline", "Poll request hit count limit. Request likely incomplete.");
+    }
+    const s = results.statuses[0].id_str;
+    this.twitter.storage.set_since(feed.hashtag, s);
+    // If req.count = 1, the resp will be the initial tweet used to get initial "since"
+    if (req.count !== 1) {
+      try {
+        yield this.twitter.processor.process_tweets(feed.room, results.statuses, 0);
       }
-      else{
-        if(results.statuses.length === HASHTAG_TWEET_FETCH_COUNT) {
-          log.info("Hashtag poll request hit count limit. Request likely incomplete.");
-        }
+      catch(err) {
+        log.error("Timeline", "Error whilst processing hashtag feed %s: %s", feed.hashtag, err);
       }
-      const s = results.statuses[0].id_str;
-      this.twitter.storage.set_since(feed.hashtag, s);
-      log.silly("Storing since: %s", s);
-      this.twitter.processor.process_tweets(feed.room, results.statuses, 0);
-    }).catch((error) => {
-      log.error("_process_hashtag_feed: GET /search/tweets returned: %s", error);
-    });
+    }
+  }
 
-    this._h++;
-    if(this._h >= this._hashtags.length) {
-      this._h = 0;
+  * _check_empty_rooms () {
+    const bot = this.twitter.bridge.getBot();
+    const memberLists = yield bot.getMemberLists();
```

Comment:
This does a `/sync`, albeit a filtered one. I guess this is OK if it's only being done every 30s.